SEED-platform/seed
# feat: add BETTER portfolio analysis

#### Any background context you want to provide?
SEED is improving BETTER integration

#### What's this PR do?
- adds ability for users to run Portfolio analysis on buildings in BETTER
- adds information tooltips to BETTER analysis creation modal
- refactoring organization of the BETTER analysis pipeline (sorry for large diff 😬 )

#### How should this be manually tested?
Upload the file below. Be sure to map the required params: City, State, Property Type, Gross floor area, ...
Also import the meter data in the same file.

From inventory list page, you should be able to run a BETTER analysis without portfolio analysis and with portfolio analysis. For portfolio analysis, you should see the HTML file for both the individual building as well as the portfolio when you visit the analysis run page. Also when running a portfolio analysis it should update the inventory list with the BETTER columns described in #2775 . Verify things still look correct when not running the portfolio analysis.

You should not be able to select the "portfolio analysis" checkbox when only one property was selected for the analysis.

[PM_BETTER_PORTFOLIO_INPUT.xlsx](https://github.com/SEED-platform/seed/files/6901495/PM_BETTER_PORTFOLIO_INPUT.xlsx)


#### What are the relevant tickets?

#2768 

#### Screenshots (if appropriate)

New checkbox for running a portfolio analysis
![Screen Shot 2021-07-28 at 7 31 12 PM](https://user-images.githubusercontent.com/18518728/127417579-17328d09-0e4b-4c2a-82f0-28574f070c3b.png)

Viewing analysis summary. Note that one property was intentionally failed by removing its "City" field. Notice that each successful property has two html files now - one for the portfolio analysis and one for the individual analysis
![Screen Shot 2021-07-28 at 7 27 31 PM](https://user-images.githubusercontent.com/18518728/127417336-273a2f78-f67e-44a8-8096-41f25ad22f32.png)

Viewing portfolio analysis results for a property
![Screen Shot 2021-07-28 at 7 28 08 PM](https://user-images.githubusercontent.com/18518728/127417343-af7f4148-ad18-4a05-88a1-7e74a07e9174.png)

Viewing building analysis results for the same property
![Screen Shot 2021-07-28 at 7 28 25 PM](https://user-images.githubusercontent.com/18518728/127417349-016206f9-83ce-41bb-ba17-ac5d0f266b81.png)
